### PR TITLE
Add DSN options to enable SQL queries on follower nodes

### PR DESCRIPTION
**Pull request by ggicci**

### CODE REVIEW

File: client/conn.go [hunk 1]
```diff
@@ -67,14 +79,41 @@ func newConn(cfg *Config) (c *conn, err error) {
...
+		c.leader = &pconn{
+			parent:  c,
+			pCaller: c.pCaller,
+		}
+	}
+
+	// choose a random follower node
+	if cfg.UseFollower && len(peers.Servers) > 1 {
+		for {
+			rand.Seed(time.Now().UnixNano())
```
Comment (xq262144) [inline client/conn.go]: It's inappropriate for client SDK to modify the golang runtime global random source seed.
I thought using client package global `randSource` is better, already initialized with `rand.New(rand.NewSource(time.Now().UnixNano()))`.

File: client/conn.go [hunk 2]
```diff
@@ -67,14 +79,41 @@ func newConn(cfg *Config) (c *conn, err error) {
...
+			node := peers.Servers[rand.Intn(len(peers.Servers))]
```
Comment (xq262144) [inline client/conn.go]: For example, this `rand.Intn` can be replaced with `randSource.Intn`

File: client/conn.go [hunk 3]
```diff
@@ -88,57 +127,74 @@ func newConn(cfg *Config) (c *conn, err error) {
...
+		if err := c.follower.startAckWorkers(workerCount); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *conn) stopAckWorkers() {
+	if c.leader != nil {
+		close(c.leader.ackCh)
```
Comment (xq262144) [inline client/conn.go]: Should be replaced with `c.leader.stopAckWorkers()`

File: client/conn.go [hunk 4]
```diff
@@ -88,57 +127,74 @@ func newConn(cfg *Config) (c *conn, err error) {
...
+	}
+	if c.follower != nil {
+		close(c.follower.ackCh)
```
Comment (xq262144) [inline client/conn.go]: Likewise, `c.follower.stopAckWorkers()`

File: client/conn.go [hunk 5]
```diff
@@ -67,14 +79,41 @@ func newConn(cfg *Config) (c *conn, err error) {
...
+		}
+	}
+
+	// choose a random follower node
+	if cfg.UseFollower && len(peers.Servers) > 1 {
+		for {
+			rand.Seed(time.Now().UnixNano())
+			node := peers.Servers[rand.Intn(len(peers.Servers))]
+			if node != peers.Leader {
+				c.pFollowerCaller = rpc.NewPersistentCaller(node)
```
Comment (xq262144) [inline client/conn.go]: Missing `CloseStream` for `c.pFollowerCaller`

**Review by xq262144 — APPROVED** [review]
LGTM

**Review by leventeliu — APPROVED** [review]
LGTM, approve.

**Pull request merged**